pagina adoção

mahmvs opened the pull request:
Closes #9

mahmvs committed:
pagina adocao

marcosdly committed:
Revert "pagina adocao"

This reverts commit 47236f6f94517c7dcbb575272bf3f9e25f7a7dc3.

marcosdly committed:
chore(frontend): add figma static image assets

marcosdly committed:
feat(adocao): animal display card

marcosdly committed:
feat(adocao): animal loading filter button

marcosdly committed:
feat(adocao): animal displaying pagination

marcosdly committed:
feat(frontend): Adocao page

marcosdly commented:
Reverti o commit único e separei as em diversos commits. Prefira micro commits.

marcosdly committed:
Merge branch 'develop' into feature/page-adocao

marcosdly commented on frontend/src/components/AdocaoCard/_adocaoCard.scss:
Faltando layout mobile.

marcosdly commented on frontend/src/components/AdocaoCard/AdocaoCard.tsx:
```diff
@@ -0,0 +1,34 @@
+interface AdocaoCardProps {
+  titulo: string;
+  idade: string;
+  id?: number;
```
Propriedade não utilizada

marcosdly replied:
Fixed c0075aee9aed718d2bda7350f6145dcaa3935670

marcosdly commented on frontend/src/components/AdocaoCard/AdocaoCard.tsx:
```diff
@@ -0,0 +1,34 @@
+      <div className="adocaocard__container__item">
+        <div className="adocaocard__container__item2">
+          <p className="adocaocard__container__item__title adocaocard__container__item__title2">
+            {' '}
```
Espaço redundante

marcosdly replied:
Fixed c0075aee9aed718d2bda7350f6145dcaa3935670

marcosdly commented on frontend/src/components/AdocaoCard/AdocaoCard.tsx:
```diff
@@ -0,0 +1,34 @@
+        <img src={imageUrl} alt="" />
+      </div>
+      <div className="adocaocard__container__item">
+        <div className="adocaocard__container__item2">
```
Classes insuficientemente descritivas

marcosdly replied:
Fixed:
- c0075aee9aed718d2bda7350f6145dcaa3935670
- eb14db6d925e89db92dd03aea271ce022fa783bb
- 27462cc7b701061531ca9d8d7498c31d9c219873

marcosdly commented on frontend/src/pages/Adocao/Adocao.tsx:
```diff
@@ -0,0 +1,72 @@
+];
+
+const Adocao: FC<unknown> = () => {
+  const [filter, setFilter] = useState('all');
```
Se apenas dois valores são possíveis, um [useReducer](https://react.dev/reference/react/useReducer) é mais eficiente do que um useState.

marcosdly replied:
Fixed 46b851efca9fc8cdaf9e3ef88638fa13e40a2b39

marcosdly commented on frontend/src/pages/Adocao/_adocao.scss:
```diff
@@ -0,0 +1,59 @@
+      align-content: flex-start;
+      align-items: flex-start;
+
+      @media screen and (max-width: 1156px) {
```
Falta de documentação em media queries incomuns que testam o tamanho da tela

marcosdly replied:
Fixed:
- e0e846e9138b4915a63f9955e40490fe0a92cae8
- 27462cc7b701061531ca9d8d7498c31d9c219873
- eb14db6d925e89db92dd03aea271ce022fa783bb

marcosdly commented on frontend/src/pages/Adocao/_adocao.scss:
```diff
@@ -0,0 +1,59 @@
+    padding-top: 6.25rem;
+
+    &__body {
+      margin-top: 11.063rem;
```
* Valores absolutos não documentados/abstraídos
* Números demasiadamente específicos

marcosdly replied:
Fixed:
- e0e846e9138b4915a63f9955e40490fe0a92cae8
- a56142860df9c7b0fb6e2046dfec6dacf9533b3f
- 27462cc7b701061531ca9d8d7498c31d9c219873
- eb14db6d925e89db92dd03aea271ce022fa783bb

marcosdly committed:
refactor(adocao): optimized AdocaoCard component

- More descriptive class names
- Removed redundant content
- Fixed non-standard typing declarations
- Removed unnecessary containers
- Removed unused props
- Fixed /public URLs

marcosdly committed:
refactor(AdocaoCard): updated class names, optimized styling

- Updated class names
- Reduced amount of layout-ing
- Reduced amount of specific spacing
- Removed redudant styling

marcosdly commented on frontend/src/components/AdocaoCard/_adocaoCard.scss:
```diff
@@ -0,0 +1,68 @@
+  }
+}
+
+.adocaocard__container {
```
> [!CAUTION]
> CSS de um componente genérico modifica o estilo de um container pai (parent)

marcosdly replied:
Fixed:
- 2f29a2393e4e2176f10592213d93c89ed5631efd
- b47d6d6351e73c48b38db1847524000e1d60f325

marcosdly committed:
feat(adocao): filter state provider with constant options

marcosdly committed:
refactor(FilterButton): using filter state provider

marcosdly committed:
refactor(Paginacao): using filter state provider

marcosdly committed:
refactor(Adocao): using filter state provider

marcosdly committed:
fix(AdocaoCard): parent CSS inside children component

marcosdly committed:
refactor(Paginacao): optimize css, remove unused code

- Convert imports of public assets to constant strings
- Remove unused properties
- More descriptive classes
- More consistent CSS properties
- More specific CSS rules

marcosdly committed:
fix(AdocaoCard): incorrect typing, generic prop isn't generic

marcosdly committed:
refactor(Adocao): generate dummy data

marcosdly committed:
refactor(Adocao): optimized css, optimized component's html

- Removed redundant elements
- More descriptive classes
- More consistent CSS properties
- More specific CSS rules

marcosdly committed:
feat(Paginacao): page buttons according to current page

marcosdly committed:
feat(Paginacao): page button designed style

marcosdly committed:
feat(Paginacao): always display last page if available

marcosdly committed:
fix(Adocao): missing bottom margin

marcosdly committed:
fix(Paginacao): snappy spacing of page select buttons

marcosdly committed:
fix(Paginacao): if-conditions don't scale with page amount

marcosdly committed:
chore(AdocaoCard): delete unused code

marcosdly commented:
⚠️ Pausando minha contribuição aqui em favor da arquitetura do site
